src/components/Toolbar/Toolbar.tsx

```diff
@@ -22,17 +23,21 @@ export enum ToolbarButtonType {
 export type ToolBarProps = {
   setActiveTool: (activeTool: ToolbarButtonType | null) => void;
   activeTool: ToolbarButtonType | null;
+  isArrowButtonDisabled: boolean;
 };
 
 export const Toolbar: React.FC<ToolBarProps> = ({
   setActiveTool,
   activeTool,
+  isArrowButtonDisabled,
 }) => {
   const [activeButton, setActiveButton] = React.useState<string | null>(
     activeTool,
   );
 
   const setActive = (newValue: ToolbarButtonType): void => {
+    if (newValue === ToolbarButtonType.CreateArrow && isArrowButtonDisabled)
+      return;
```

Comment:
Do we need this when we use the `disabled` attribute?

Reply:
Indeed, forgot to remove it. Thanks!